app/controllers/admin/delivery_partners_controller.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+class Admin::DeliveryPartnersController < Admin::BaseController
+  skip_after_action :verify_authorized
+  skip_after_action :verify_policy_scoped
+
+  def choose_lead_providers
+    authorize DeliveryPartner, :create?
+
+    new_supplier_form = NewSupplierForm.new(session[:new_supplier_form])
+    if session[:delivery_partner_form].nil?
+      session[:delivery_partner_form] = { name: new_supplier_form.name }
+    else
+      session[:delivery_partner_form].merge!({ name: new_supplier_form.name })
+    end
+
+    @delivery_partner_form = DeliveryPartnerForm.new(session[:delivery_partner_form])
+  end
+
+  def receive_lead_providers
+    authorize DeliveryPartner, :create?
+
+    provider_relationships = []
+    # Ensure all selected lead providers have at least one selected cohort
+    chosen_lead_providers = params.dig(:delivery_partner_form, :lead_providers)&.keep_if(&:present?)
+
+    unless chosen_lead_providers&.any?
+      @delivery_partner_form = DeliveryPartnerForm.new(session[:delivery_partner_form])
+      @delivery_partner_form.errors.add(:lead_providers, :blank, message: "Choose at least one")
+      render :choose_lead_providers and return
+    end
+
+    chosen_lead_providers.each do |lead_provider|
+      chosen_cohorts = params.dig(
+        :delivery_partner_form,
+        lead_provider.to_sym,
+        :cohorts,
+      )&.keep_if(&:present?)
+
+      unless chosen_cohorts&.any?
+        @delivery_partner_form = DeliveryPartnerForm.new(session[:delivery_partner_form])
+        @delivery_partner_form.errors.add(:lead_providers, :blank, message: "Choose at least one cohort for every selected lead provider")
+        # rubocop:disable Lint/NonLocalExitFromIterator
+        render :choose_lead_providers and return
+        # rubocop:enable Lint/NonLocalExitFromIterator
+      end
```

Comment:
Again, can we move this validation to the `DeliveryPartnerForm` class? Probably means we will not need to disable rubocop either.

Reply:
How would we do that? Let's say I'm at step one of a three step form, and all three parts are required on the form. I don't know how to check validity

Reply:
I think you can use `on` for this.

`apply-for-teacher-training` has things like
```
validates :award_year, presence: true, on: :award_year
...
validates :grade, length: { maximum: 6 }, on: :grade, unless: :international_gcses_flag_active?
...
validates :award_year, presence: true, on: :details
```
which look like validation that is only active on some pages / urls / controller methods. 
```
  def then_i_see_validation_errors_for_my_qualification
    expect(page).to have_content t('activemodel.errors.models.candidate_interface/other_qualification_wizard.attributes.award_year.blank')
  end
```
looks like a test for one of them.

Reply:
If that fails, or is too deep of a rabbit hole, well, we can keep track of the step and update it in session each time we go to a controller method - and that's how the form can know what to validate.

Reply:
This looks like it will work 🙂 It takes `on: :context` and then you call `.valid?(:context)` which only checks matching contexts